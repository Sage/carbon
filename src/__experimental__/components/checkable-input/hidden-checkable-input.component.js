import React from 'react';
import PropTypes from 'prop-types';
import HiddenCheckableInputStyle from './hidden-checkable-input.style';

const HiddenCheckableInput = ({
  checked, role, type, ...props
}) => {
  return (
    <HiddenCheckableInputStyle
      aria-checked={ checked }
      role={ role || type }
      type={ type }
      { ...props }
    />
  );
};

HiddenCheckableInput.propTypes = {
  checked: PropTypes.bool,
  role: PropTypes.string,
  type: PropTypes.string.isRequired
};

export default HiddenCheckableInput;
