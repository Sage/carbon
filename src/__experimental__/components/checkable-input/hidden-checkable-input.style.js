import styled from 'styled-components';

const HiddenCheckableInputStyle = styled.input`
  cursor: pointer;
  opacity: 0;
  margin: 0;
  z-index: 999;
`;

export default HiddenCheckableInputStyle;
