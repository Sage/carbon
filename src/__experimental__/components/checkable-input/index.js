export { default } from './checkable-input.component';
