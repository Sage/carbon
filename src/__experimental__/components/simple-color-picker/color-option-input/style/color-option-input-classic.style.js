import { css } from 'styled-components';
import { THEMES } from '../../../../../style/themes';
import StyledColorSampleBox from '../../color-sample-box/style/color-sample-box.style';

export default ({ theme }) => theme.name === THEMES.classic && css`
  &:checked + ${StyledColorSampleBox}, 
  &:focus + ${StyledColorSampleBox} {
    box-shadow: none;
    border: 1px solid #003349;
  }
`;
