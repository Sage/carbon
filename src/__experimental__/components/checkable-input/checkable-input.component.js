import React from 'react';
import PropTypes from 'prop-types';
import { validProps } from '../../../utils/ether';
import { StyledCheckableInput, StyledCheckableInputWrapper } from './checkable-input.style';
import FormField from '../form-field';
import HiddenCheckableInput from './hidden-checkable-input.component';
import guid from '../../../utils/helpers/guid';

class CheckableInput extends React.Component {
  constructor(props) {
    super(props);
    this.inputId = props.inputId || guid();
  }

  render() {
    const {
      children, onChange, onBlur, ...rest
    } = this.props;
    const id = this.inputId;
    const labelId = `${id}-label`;
    const helpId = `${id}-help`;

    const formFieldProps = {
      ...validProps(this, ['fieldHelp', 'fieldHelpInline', 'labelHelp', 'reverse']),
      labelId,
      helpId,
      label: rest.inputLabel,
      labelHelpIcon: 'info',
      name: id
    };

    const {
      fieldHelp, labelHelp, ...inputProps
    } = {
      ...validProps(this, ['checked', 'disabled', 'inputType', 'onChange', 'onBlur', 'tabindex']),
      labelId,
      helpId,
      id
    };

    return (
      <StyledCheckableInputWrapper { ...rest }>
        <FormField { ...formFieldProps }>
          <StyledCheckableInput>
            <HiddenCheckableInput { ...inputProps } />
            {children}
          </StyledCheckableInput>
        </FormField>
      </StyledCheckableInputWrapper>
    );
  }
}

CheckableInput.propTypes = {
  /** Set the value of the CheckableInput */
  checked: PropTypes.bool,
  /** Used to set the visible aspect of the input (i.e. the checkbox sprite, input slider etc) */
  children: PropTypes.node,
  /** Toggles disabling of input */
  disabled: PropTypes.bool,
  /** Toggles error styles */
  error: PropTypes.bool,
  /** The fieldHelp content to display for the input */
  fieldHelp: PropTypes.node,
  /** Displays fieldHelp inline with the CheckableInput */
  fieldHelpInline: PropTypes.bool,
  /** Unique Identifier for the input. Will use a randomly generated GUID if none is provided */
  inputId: PropTypes.string,
  /** The content for the Label to apply to the input */
  inputLabel: PropTypes.node,
  /** Sets percentage-based input width */
  inputWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Sets label alignment - accepted values: 'left' (default), 'right' */
  labelAlign: PropTypes.string,
  /** The content for the help tooltip, to appear next to the Label */
  labelHelp: PropTypes.node,
  /** Sets percentage-based label width */
  labelWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Accepts a callback function which can be used to update parent state on change */
  onChange: PropTypes.func,
  /** Accepts a callback function which is triggered on blur event */
  onBlur: PropTypes.func,
  /** Reverses label and CheckableInput display */
  reverse: PropTypes.bool,
  /** Specifies input type, 'checkbox' or 'switch' */
  inputType: PropTypes.string.isRequired
};

CheckableInput.defaultProps = {
  reverse: false
};

export default CheckableInput;
