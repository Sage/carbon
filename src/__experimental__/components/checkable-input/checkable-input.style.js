import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import FieldHelpStyle from '../field-help/field-help.style';
import { FieldLineStyle } from '../form-field/form-field.style';
import HiddenCheckableInputStyle from './hidden-checkable-input.style';
import LabelStyle from '../label/label.style';
import StyledHelp from '../../../components/help/help.style';
import baseTheme from '../../../style/themes/base';
import StyledValidationIcon from '../../../components/validations/validation-icon.style';

const StyledCheckableInput = styled.div`
  display: inline-block;
  position: relative;
`;

const StyledCheckableInputWrapper = styled.div`
  ${({
    disabled, fieldHelpInline, inputWidth, labelAlign, labelWidth, reverse, theme
  }) => css`
    ${FieldLineStyle} {
      display: flex;
    }

    ${LabelStyle} {
      text-align: ${labelAlign};
      padding-top: 0;
      width: auto;

      & ${StyledHelp},
      & ${StyledValidationIcon} {
        color: ${theme.help.color};
        vertical-align: middle;
        top: -1px;

        &:hover, &:focus {
          color: ${theme.text.color};
        }
      }
    }

    ${FieldHelpStyle} {
      flex-basis: 100%;
    }

    ${disabled && css`
      ${LabelStyle} {
        &, & ${StyledHelp} {
          color: ${theme.disabled.disabled};
        }
      }

      ${HiddenCheckableInputStyle},
      ${LabelStyle} {
        &:hover, &:focus {
          outline: none;
          cursor: not-allowed;
        }
      }
    `}

    ${fieldHelpInline && css`
      ${FieldLineStyle} {
        flex-wrap: nowrap;
      }

      ${StyledCheckableInput} {
        margin-right: 0;
        margin-left: 8px;
      }
      
      ${FieldHelpStyle} {
        flex-grow: 0;
        flex-basis: auto;
        padding-left: 0;
        width: auto;
      }
    `}

    ${reverse && fieldHelpInline && css`
      ${StyledCheckableInput} {
        margin-left: 0;
      }

      ${FieldHelpStyle} {
        flex-grow: 1;
      }
    `}

    ${inputWidth !== undefined && inputWidth !== 0 && css`
      ${StyledCheckableInput} {
        width: ${inputWidth}% !important;
      }
    `}

    ${labelWidth !== undefined && labelWidth !== 0 && `
      ${LabelStyle} {
        width: ${labelWidth}% !important;
      }
    `}
  `}
`;

StyledCheckableInputWrapper.propTypes = {
  disabled: PropTypes.bool,
  fieldHelpInline: PropTypes.bool,
  inputWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  labelAlign: PropTypes.string,
  labelWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  theme: PropTypes.object
};

StyledCheckableInputWrapper.defaultProps = {
  labelAlign: 'left',
  theme: baseTheme
};

export { StyledCheckableInput, StyledCheckableInputWrapper };
