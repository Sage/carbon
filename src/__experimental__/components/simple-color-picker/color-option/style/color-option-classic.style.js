import { css } from 'styled-components';
import { THEMES } from '../../../../../style/themes';

export default ({ theme }) => theme.name === THEMES.classic
  && css`
    margin-right: 1px;
    margin-bottom: 1px;
  `;
