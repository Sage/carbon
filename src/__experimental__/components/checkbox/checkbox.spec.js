import React from 'react';
import TestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { css } from 'styled-components';
import Checkbox from './checkbox.component';
import { StyledCheckableInput } from '../checkable-input/checkable-input.style';
import FieldHelpStyle from '../field-help/field-help.style';
import HiddenCheckableInputStyle from '../checkable-input/hidden-checkable-input.style';
import LabelStyle from '../label/label.style';
import StyledCheckableInputSvgWrapper from '../checkable-input/checkable-input-svg-wrapper.style';
import StyledHelp from '../../../components/help/help.style';
import guid from '../../../utils/helpers/guid';
import { assertStyleMatch, carbonThemesJestTable } from '../../../__spec_helper__/test-utils';
import { baseTheme, classicTheme } from '../../../style/themes';

jest.mock('../../../utils/helpers/guid');
guid.mockImplementation(() => 'guid-12345');

function render(props, renderer = TestRenderer.create, options = {}) {
  return renderer(
    <Checkbox
      name='my-checkbox'
      value='test'
      { ...props }
    />,
    options
  );
}

describe('Checkbox', () => {
  describe('base theme', () => {
    it('renders as expected', () => {
      expect(render()).toMatchSnapshot();
    });

    describe('when size=large', () => {
      let wrapper;

      beforeEach(() => {
        wrapper = render({ size: 'large' }).toJSON();
      });

      it('applies the appropriate input display element styles', () => {
        const styles = {
          height: '24px',
          width: '24px'
        };

        assertStyleMatch(styles, wrapper, { modifier: css`${StyledCheckableInput}` });

        assertStyleMatch(styles, wrapper, { modifier: css`${HiddenCheckableInputStyle}` });

        assertStyleMatch(styles, wrapper, { modifier: 'svg' });

        assertStyleMatch({
          height: '24px'
        }, wrapper, { modifier: (css`${StyledCheckableInputSvgWrapper}`) });
      });

      it('applies the correct CheckboxSvgWrapper styles', () => {
        assertStyleMatch({
          height: '24px'
        }, wrapper, { modifier: (css`${StyledCheckableInputSvgWrapper}`) });
      });

      it('applies the appropriate FieldHelp styles', () => {
        assertStyleMatch({
          marginLeft: '24px'
        }, wrapper, { modifier: css`${FieldHelpStyle}` });
      });

      describe('when labelSpacing is 2', () => {
        it('should apply the correct fieldHelp styles', () => {
          wrapper = render({ labelSpacing: 2, size: 'large' }).toJSON();
          assertStyleMatch({
            paddingLeft: '16px',
            marginLeft: '24px'
          }, wrapper, { modifier: css`${FieldHelpStyle}` });
        });
      });
    });

    describe('when size=large and fieldHelpInline=true', () => {
      const wrapper = render({ fieldHelpInline: true, size: 'large' }).toJSON();

      it('applies the appropriate FieldHelp styles', () => {
        assertStyleMatch({
          marginTop: '0'
        }, wrapper, { modifier: css`${FieldHelpStyle}` });
      });
    });

    describe('when checkbox is checked', () => {
      it('renders the correct check colour', () => {
        const wrapper = render({ checked: true }).toJSON();

        assertStyleMatch({
          fill: baseTheme.checkable.checked
        }, wrapper, {
          modifier: css`${HiddenCheckableInputStyle}:checked ~ ${StyledCheckableInputSvgWrapper} svg path`
        });
      });

      describe('and disabled=true', () => {
        const wrapper = render({ checked: true, disabled: true }).toJSON();

        it('renders the correct check colour', () => {
          assertStyleMatch({
            fill: baseTheme.disabled.border
          }, wrapper, {
            modifier: css`${HiddenCheckableInputStyle}:checked ~ ${StyledCheckableInputSvgWrapper} svg path`
          });
        });
      });
    });

    describe('when disabled=true', () => {
      const wrapper = render({ disabled: true }).toJSON();

      it('applies the appropriate svg wrapper styles', () => {
        assertStyleMatch({
          backgroundColor: baseTheme.disabled.input,
          border: `1px solid ${baseTheme.disabled.border}`
        }, wrapper, { modifier: 'svg' });
      });

      it('applies the appropriate check styles', () => {
        assertStyleMatch({
          fill: baseTheme.disabled.input
        }, wrapper, { modifier: 'svg path' });
      });

      describe('and hover / focus is applied', () => {
        const hoverFocusStyles = {
          outline: 'none',
          cursor: 'not-allowed'
        };

        it('applies the appropriate svg hover styles', () => {
          assertStyleMatch(hoverFocusStyles, wrapper, { modifier: css`${`${StyledCheckableInputSvgWrapper}:hover`}` });
        });

        it('applies the appropriate svg focus styles', () => {
          assertStyleMatch(hoverFocusStyles, wrapper, { modifier: css`${`${StyledCheckableInputSvgWrapper}:focus`}` });
        });
      });
    });

    describe('when using validation props', () => {
      let wrapper;

      beforeEach(() => {
        wrapper = mount((
          <Checkbox
            name='checkbox-warning'
            value='my-value'
          />
        ));
      });

      describe.each(['error', 'warning', 'info'])('when %s is true', (type) => {
        it('show correct border on radio', () => {
          wrapper.setProps({ [type]: true });
          const borderWidth = type === 'error' ? 2 : 1;
          assertStyleMatch({
            border: `${borderWidth}px solid ${baseTheme.colors[type]}`
          }, wrapper, { modifier: 'svg' });
        });
      });

      describe.each(['error', 'warning', 'info'])('when %s is "string', (type) => {
        it('show correct border on radio', () => {
          wrapper.setProps({ [type]: 'Message' });
          const borderWidth = type === 'error' ? 2 : 1;
          assertStyleMatch({
            border: `${borderWidth}px solid ${baseTheme.colors[type]}`
          }, wrapper, { modifier: 'svg' });
        });
      });

      describe('when error is true', () => {
        it('render correct color for errors', () => {
          wrapper.setProps({
            error: true
          });

          assertStyleMatch({
            border: `2px solid ${baseTheme.colors.error}`
          }, wrapper, { modifier: 'svg' });
        });
      });

      describe('when warning is true', () => {
        it('render correct color for warnings', () => {
          wrapper.setProps({
            warning: true
          });

          assertStyleMatch({
            border: `1px solid ${baseTheme.colors.warning}`
          }, wrapper, { modifier: 'svg' });
        });
      });

      describe('when info is true', () => {
        it('render correct color for info', () => {
          wrapper.setProps({
            info: true
          });

          assertStyleMatch({
            border: `1px solid ${baseTheme.colors.info}`
          }, wrapper, { modifier: 'svg' });
        });
      });
    });

    describe('when fieldHelpInline is true', () => {
      it('renders the correct FieldHelp styles', () => {
        const wrapper = render({ fieldHelpInline: true }).toJSON();

        assertStyleMatch({ marginLeft: '0' }, wrapper, { modifier: css`${FieldHelpStyle}` });
      });
    });

    describe('when setting a custom inputWidth', () => {
      describe('default', () => {
        const wrapper = render({ inputWidth: 50 }).toJSON();

        it('renders the correct FieldHelp styles', () => {
          assertStyleMatch({
            marginLeft: '50% !important'
          }, wrapper, { modifier: css`${FieldHelpStyle}` });
        });
      });

      describe('reversed', () => {
        it('renders the correct FieldHelp styles', () => {
          const wrapper = render({ inputWidth: 50, reverse: true }).toJSON();

          assertStyleMatch({
            marginRight: '50% !important'
          }, wrapper, { modifier: css`${FieldHelpStyle}` });
        });
      });
    });

    describe('when reverse is true', () => {
      describe('default', () => {
        const wrapper = render({ reverse: true }).toJSON();

        it('renders the correct FieldHelp styles', () => {
          assertStyleMatch({
            marginLeft: '0'
          }, wrapper, { modifier: css`${FieldHelpStyle}` });
        });
      });

      describe('and fieldHelpInline is true', () => {
        const wrapper = render({ reverse: true, fieldHelpInline: true }).toJSON();

        it('renders the correct CheckableInput styles', () => {
          assertStyleMatch({
            marginRight: '8px'
          }, wrapper, { modifier: css`${StyledCheckableInput}` });
        });
      });
    });

    describe('when labelSpacing is 2', () => {
      const wrapper = render({ labelSpacing: 2 }).toJSON();
      assertStyleMatch({
        paddingLeft: '16px',
        marginLeft: '16px'
      }, wrapper, { modifier: css`${FieldHelpStyle}` });
    });
  });

  describe('Classic theme', () => {
    const opts = { theme: classicTheme };

    describe('default', () => {
      const wrapper = render(opts).toJSON();

      it('applies appropriate CheckableInput styles', () => {
        assertStyleMatch({
          height: '15px',
          padding: '1px 0 0 0',
          width: '15px'
        }, wrapper, { modifier: css`${StyledCheckableInput}` });
      });

      it('applies appropriate hidden input styles', () => {
        assertStyleMatch({
          height: '15px',
          padding: '1px',
          width: '15px'
        }, wrapper, { modifier: css`${HiddenCheckableInputStyle}` });
      });

      it('applies appropriate svg styles', () => {
        assertStyleMatch({
          height: '15px',
          width: '15px'
        }, wrapper, { modifier: 'svg' });
      });

      it('applies appropriate FieldHelp styles', () => {
        assertStyleMatch({
          marginLeft: '15px',
          paddingLeft: '5px'
        }, wrapper, { modifier: css`${FieldHelpStyle}` });
      });

      it('applies appropriate Label styles', () => {
        assertStyleMatch({
          paddingBottom: '0',
          paddingTop: '0'
        }, wrapper, { modifier: css`${LabelStyle}` });
      });

      describe('Help styles', () => {
        it('applies correct default style', () => {
          assertStyleMatch({
            color: '#8099a4'
          }, wrapper, { modifier: css`${LabelStyle} ${StyledHelp}` });
        });

        it('applies correct hover style', () => {
          assertStyleMatch({
            color: '#8099a4'
          }, wrapper, { modifier: css`${`${LabelStyle} ${StyledHelp}:hover`}` });
        });

        it('applies correct focus style', () => {
          assertStyleMatch({
            color: '#8099a4'
          }, wrapper, { modifier: css`${`${LabelStyle} ${StyledHelp}:focus`}` });
        });
      });

      describe('when hover / focus is applied', () => {
        it('applies the appropriate hidden input svg wrapper hover styles', () => {
          assertStyleMatch(
            { boxShadow: 'none' },
            wrapper,
            {
              modifier: css`${`${HiddenCheckableInputStyle}:not([disabled]):focus + ${StyledCheckableInputSvgWrapper}`}`
            }
          );
        });

        it('applies the appropriate hidden input svg wrapper focus styles', () => {
          assertStyleMatch(
            { boxShadow: 'none' },
            wrapper,
            {
              modifier: css`${`${HiddenCheckableInputStyle}:not([disabled]):hover + ${StyledCheckableInputSvgWrapper}`}`
            }
          );
        });

        it('applies the appropriate hidden input svg hover styles', () => {
          assertStyleMatch({
            border: '1px solid #1963f6',
            outline: 'none'
          },
          wrapper,
          {
            modifier: css`
              ${`${HiddenCheckableInputStyle}:not([disabled]):focus + ${StyledCheckableInputSvgWrapper} > svg`}
            `
          });
        });

        it('applies the appropriate hidden input svg focus styles', () => {
          assertStyleMatch({
            border: '1px solid #1963f6',
            outline: 'none'
          },
          wrapper,
          {
            modifier: css`
              ${`${HiddenCheckableInputStyle}:not([disabled]):hover + ${StyledCheckableInputSvgWrapper} > svg`}
            `
          });
        });
      });
    });

    describe('when checked is true', () => {
      it('renders the correct check colour', () => {
        const wrapper = render({ checked: true, ...opts }).toJSON();

        assertStyleMatch({
          fill: 'rgba(0,0,0,0.85)'
        }, wrapper, {
          modifier: css`${HiddenCheckableInputStyle}:checked ~ ${StyledCheckableInputSvgWrapper} svg path`
        });
      });

      describe('and disabled is true', () => {
        const wrapper = render({ checked: true, disabled: true, ...opts }).toJSON();

        it('renders the correct check colour', () => {
          assertStyleMatch({
            fill: '#8099a4'
          }, wrapper, {
            modifier: css`${HiddenCheckableInputStyle}:checked ~ ${StyledCheckableInputSvgWrapper} svg path`
          });
        });
      });
    });

    describe('when disabled is true', () => {
      const wrapper = render({ disabled: true, ...opts }).toJSON();

      it('applies the appropriate Label styles', () => {
        assertStyleMatch({
          color: classicTheme.disabled.text
        }, wrapper, { modifier: css`${LabelStyle}` });
      });

      it('applies the appropriate Help styles', () => {
        assertStyleMatch({
          color: '#8099a4'
        }, wrapper, { modifier: css`${LabelStyle} ${StyledHelp}` });
      });

      it('applies the appropriate svg wrapper styles', () => {
        assertStyleMatch({
          backgroundColor: '#e6ebed',
          border: `1px solid ${classicTheme.disabled.border}`
        }, wrapper, { modifier: 'svg' });
      });

      it('applies the appropriate check styles', () => {
        assertStyleMatch({
          fill: '#e6ebed'
        }, wrapper, { modifier: 'svg path' });
      });
    });

    describe('when fieldHelpInline is true', () => {
      it('applies the appropriate FieldHelp style', () => {
        const wrapper = render({ fieldHelpInline: true, ...opts }).toJSON();

        assertStyleMatch({
          marginLeft: '0',
          paddingBottom: '0',
          paddingTop: '0'
        }, wrapper, { modifier: css`${FieldHelpStyle}` });
      });
    });

    describe.each(carbonThemesJestTable)('when the theme is set to %s', (themeName, theme) => {
      it('sets the appropriate check colour', () => {
        const wrapper = render({ theme, checked: true }).toJSON();

        assertStyleMatch({
          fill: theme.checkable.checked
        }, wrapper, {
          modifier: css`${HiddenCheckableInputStyle}:checked ~ ${StyledCheckableInputSvgWrapper} svg path`
        });
      });
    });
  });
});
