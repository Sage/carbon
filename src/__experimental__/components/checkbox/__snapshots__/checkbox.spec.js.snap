// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox base theme renders as expected 1`] = `
.c3 + .c2 {
  margin-top: 16px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  cursor: pointer;
  opacity: 0;
  margin: 0;
  position: relative;
  z-index: 999;
}

.c7 {
  display: inline-block;
  position: relative;
}

.c11 .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 .c4 {
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c11 .c6 {
  margin-right: 0;
  margin-left: 8px;
}

.c11 .c6 {
  margin-left: 0;
}

.c12 .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 .c8:hover,
.c12 .sc-ifAKCX:hover,
.c12 .c8:focus,
.c12 .sc-ifAKCX:focus {
  outline: none;
  cursor: not-allowed;
}

.c13 .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 .c6 {
  width: 50% !important;
}

.c14 .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15 .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c15 .c4 {
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c15 .c6 {
  margin-right: 0;
  margin-left: 8px;
}

.c1 .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .c25 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-top: 0;
  width: auto;
}

.c1 .c25 .c26,
.c1 .c25 .c27 {
  color: rgba(0,0,0,0.65);
  vertical-align: middle;
}

.c1 .c25 .c26:hover,
.c1 .c25 .c27:hover,
.c1 .c25 .c26:focus,
.c1 .c25 .c27:focus {
  color: rgba(0,0,0,0.9);
}

.c1 .c24 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c16 .c6 {
  padding-top: 1px;
}

.c16 .c10 {
  height: 16px;
}

.c16 .c8,
.c16 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c16 .c6,
.c16 .c8,
.c16 .c10,
.c16 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c16 .c8:not([disabled]):focus + .c10,
.c16 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c16 .c10 {
  height: 24px;
}

.c16 .c6,
.c16 .c8,
.c16 .c10,
.c16 svg {
  height: 24px;
  width: 24px;
}

.c16 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c17 .c6 {
  padding-top: 1px;
}

.c17 .c10 {
  height: 16px;
}

.c17 .c8,
.c17 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c17 .c6,
.c17 .c8,
.c17 .c10,
.c17 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c17 .c8:not([disabled]):focus + .c10,
.c17 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c17 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c17 .c8:checked ~ .c10 svg path {
  fill: #CCD6DB;
}

.c17 .c10:hover,
.c17 .c10:focus {
  outline: none;
  cursor: not-allowed;
}

.c18 .c6 {
  padding-top: 1px;
}

.c18 .c10 {
  height: 16px;
}

.c18 .c8,
.c18 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c18 .c6,
.c18 .c8,
.c18 .c10,
.c18 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c18 .c8:not([disabled]):focus + .c10,
.c18 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c18 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c19 .c6 {
  padding-top: 1px;
}

.c19 .c10 {
  height: 16px;
}

.c19 .c8,
.c19 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c19 .c6,
.c19 .c8,
.c19 .c10,
.c19 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c19 .c8:not([disabled]):focus + .c10,
.c19 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c19 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c20 .c6 {
  padding-top: 1px;
}

.c20 .c10 {
  height: 16px;
}

.c20 .c8,
.c20 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c20 .c6,
.c20 .c8,
.c20 .c10,
.c20 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c20 .c8:not([disabled]):focus + .c10,
.c20 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c20 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c20 .c6 {
  margin-right: 8px;
}

.c21 .c6 {
  padding-top: 1px;
}

.c21 .c10 {
  height: 16px;
}

.c21 .c8,
.c21 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c21 .c6,
.c21 .c8,
.c21 .c10,
.c21 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c21 .c8:not([disabled]):focus + .c10,
.c21 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c21 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c22 .c6 {
  padding-top: 1px;
}

.c22 .c10 {
  height: 16px;
}

.c22 .c8,
.c22 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c22 .c6,
.c22 .c8,
.c22 .c10,
.c22 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c22 .c8:not([disabled]):focus + .c10,
.c22 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c22 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c22 .c6 {
  padding: 1px 0 0 0;
}

.c22 .c8,
.c22 svg {
  padding: 1px;
}

.c22 .c6,
.c22 .c8,
.c22 .c10,
.c22 svg {
  height: 15px;
  width: 15px;
}

.c22 .c8:not([disabled]):focus + .c10,
.c22 .c8:not([disabled]):hover + .c10 {
  box-shadow: none;
}

.c22 .c8:not([disabled]):focus + .c10 > svg,
.c22 .c8:not([disabled]):hover + .c10 > svg {
  border: 1px solid #1963f6;
  outline: none;
}

.c22 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.85);
}

.c23 .c6 {
  padding-top: 1px;
}

.c23 .c10 {
  height: 16px;
}

.c23 .c8,
.c23 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c23 .c6,
.c23 .c8,
.c23 .c10,
.c23 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c23 .c8:not([disabled]):focus + .c10,
.c23 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c23 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

.c23 .c8:checked ~ .c10 svg path {
  fill: #4d7080;
}

.c23 .c10:hover,
.c23 .c10:focus {
  outline: none;
  cursor: not-allowed;
}

.c23 .c6 {
  padding: 1px 0 0 0;
}

.c23 .c8,
.c23 svg {
  padding: 1px;
}

.c23 .c6,
.c23 .c8,
.c23 .c10,
.c23 svg {
  height: 15px;
  width: 15px;
}

.c23 .c8:not([disabled]):focus + .c10,
.c23 .c8:not([disabled]):hover + .c10 {
  box-shadow: none;
}

.c23 .c8:not([disabled]):focus + .c10 > svg,
.c23 .c8:not([disabled]):hover + .c10 > svg {
  border: 1px solid #1963f6;
  outline: none;
}

.c23 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.85);
}

.c23 .c8:checked ~ .c10 svg path {
  fill: #8099a4;
}

.c0 .c6 {
  padding-top: 1px;
}

.c0 .c10 {
  height: 16px;
}

.c0 svg {
  background-color: #FFFFFF;
  border: 1px solid #668592;
}

.c0 .c8,
.c0 svg {
  height: 16px;
  position: absolute;
  padding: 1px;
}

.c0 .c6,
.c0 .c8,
.c0 .c10,
.c0 svg {
  box-sizing: border-box;
  min-width: 16px;
  width: 16px;
}

.c0 .c8:not([disabled]):focus + .c10,
.c0 .c8:not([disabled]):hover + .c10 {
  box-shadow: 0 0 0 3px #FFB500;
}

.c0 .c25 {
  width: auto;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c0 .c24 {
  margin-left: 16px;
  margin-top: 0;
  padding-left: 8px;
}

.c0 .c27 {
  position: relative;
  display: inline-block;
}

.c0 .c8:checked ~ .c10 svg path {
  fill: rgba(0,0,0,0.90);
}

<div
  className="c0"
  data-component="checkbox"
  name="my-checkbox"
>
  <div
    className="c1"
    name="my-checkbox"
  >
    <div
      className="c2 c3"
    >
      <div
        className="c4 c5"
      >
        <div
          className="c6 c7"
        >
          <input
            aria-describedby="guid-12345-help"
            aria-labelledby="guid-12345-label"
            className="c8 c9"
            id="guid-12345"
            name="my-checkbox"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="checkbox"
            type="checkbox"
            value="test"
          />
          <div
            className="c10 "
          >
            <svg
              focusable="false"
              height="10"
              viewBox="0 0 12 10"
              width="12"
            >
              <path
                d="M.237 6.477A.752.752 0 0 1 .155 5.47l.851-1.092a.63.63 0 0 1 .934-.088l2.697 1.964, 4.674-6a.63.63 0 0 1 .933-.088l1.015.917c.28.254.317.703.081 1.005L6.353 8.492a.725.725, 0 0 1-.095.16l-.85 1.093a.637.637 0 0 1-.626.244.638.638 0 0 1-.335-.16L.237 6.476z"
                fill="#FFFFFF"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
